arb_os/evmOps.mini

```diff
@@ -309,34 +312,19 @@ public impure func evmOp_extcodecopy(addrAsUint: uint, memAddr: uint, codeOffset
     let addr = address(addrAsUint);     // truncates if necessary
     if let Some(topFrame) = evmCallStack_topFrame() {
         let memory = evmCallFrame_getMemory(topFrame);
-        if let Some(code) = account_getCode(
-            evmCallFrame_getAccount(topFrame, addr)
-        ) {
-            // if this would read beyond end of code,
-            //       bytearray_copy will zero-fill the rest of the target memory region,
-            //       which is the desired behavior
-            let updatedMemory = bytearray_copy(
-                code,
-                codeOffset,
-                memory,
-                memAddr,
-                nbytes
-            );
-            if (evmCallStack_setTopFrameMemory(updatedMemory,)) {
-                return;
-            }
+        let (codeArr, codeSize) = xif (evmCallFrame_constructorIsOnStackForAddress(topFrame, addr)) {
```

Comment:
I believe `codeSize` is actually an offset, not a size, and that we should have the result of the if statement just be `codeArr` and use the `codeOffset` parameter as-is. I'm assuming reading from a non-zero offset into a zero array is just zeroes.

Reply:
Good idea; done.